### PR TITLE
[Op] Update reduce_scatter with input tensor

### PR DESCRIPTION
**Pull request by StrongSpoon**
## Description ##
This pr enables op reduce_scatter to receive tensor in addition to tensor list as input.

Input tensor is split into n parts at axis 0 (n is the size of group). Part i of the tensor is reduced by rank i.

## Checklist ##

- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

cc @awslabs/raf-reviewer

### CODE REVIEW

File: python/raf/distributed/op.py [hunk 1]
```diff
@@ -131,8 +131,8 @@ def reduce_scatter(x, computation="sum", rank_list=None):
 
     Parameters
     ----------
-    x : List[Tensor]
-        A list of tensors of equal shape
+    x : Tensor | List[Tensor]
```
Comment (comaniac) [inline python/raf/distributed/op.py:134]: Please follow the existing type format in this file.
```suggestion
    x : Tensor or List[Tensor]
```

File: python/raf/distributed/op.py [hunk 2]
```diff
@@ -151,7 +151,25 @@ def reduce_scatter(x, computation="sum", rank_list=None):
...
+        for group in rank_list:
+            if comm.rank in group:
+                size = len(group)
+                break
+        else:
+            size = 1
+    else:
+        size = comm.size
+
+    if not is_list:
```
Comment (comaniac) [inline python/raf/distributed/op.py:167]: Seems not necessary to have `is_list` as it only used once.

```suggestion
    if not isinstance(x, (tuple, list)):
```

File: tests/python/distributed/test_collective_communication.py [hunk 3]
```diff
@@ -386,6 +386,61 @@ def forward(self, x, y):
                 check(m_out, odd_out)
 
 
+@pytest.mark.skipif(skip_dist_test(min_rank_num=2, require_exact_rank=True), reason=SKIP_REASON)
+@pytest.mark.parametrize("computation", ["sum", "prod", "min", "max"])
+def test_reduce_scatter_input_tensor(computation):
```
Comment (comaniac) [inline tests/python/distributed/test_collective_communication.py:391]: Make the name more informative.

```suggestion
def test_reduce_scatter_input_tensor_list(computation):
```

File: tests/python/distributed/test_collective_communication.py [hunk 4]
```diff
@@ -386,6 +386,61 @@ def forward(self, x, y):
...
+        elif computation == "min":
+            n_out = n_ones * min(1, total_rank)
+        elif computation == "max":
+            n_out = n_ones * max(1, total_rank)
+        elif computation == "avg":
+            n_out = n_ones * sum(range(1, total_rank + 1))
+            n_out = n_out / total_rank
+        else:
+            assert False, "Invalid computation"
+        check(m_out, n_out)
```
Comment (comaniac) [inline tests/python/distributed/test_collective_communication.py:428]: Seems only need one `check(m_out, n_out)` for both ranks.

**Review by comaniac — COMMENTED** [review]
Otherwise LGTM. Also please fix the CI.

**Comment by zhen-jia** [comment]
The modification inserts `split` op for a single tensor in `reduce_scatter`. Could you remind me what's the purpose of doing that? `NCCL` already supports that functionality.

**Comment by comaniac** [comment]
@StrongSpoon please rebase and do not change the 3rdparty commit in this PR.

**Pull request closed without merging**